CONTRIBUTING.rst

```diff
@@ -0,0 +1,160 @@
+
```

Comment:
This looks like an exact copy of the Cyclus Developer Workflow document (which I made some comments on) - should we just link to that document instead?